pkg/test/case.go

```diff
@@ -42,6 +43,11 @@ type Case struct {
 
 // DeleteNamespace deletes a namespace in Kubernetes after we are done using it.
 func (t *Case) DeleteNamespace(namespace string) error {
```

Comment:
It is really confusing to a casual reader right now is that we have the `namespace` arg and `t.Namespace`. They match usually, but not when `t.Namespace` has the sentinel value. We document the design in comments, which is fragile.

I think we should aim for self-documenting code whose semantics is obvious. How about making `Namespace` a `struct` with `Name string` and `Autocreated bool`. In that case we can get rid of the `namespace` arg altogether.

Reply:
Same for `CreateNamespace`

Reply:
I think I like the struct suggestion.. but requires some thinking... My initial thought is I'm expecting some refactoring shortly and perhaps it should go into that work.  If we don't do it here for this PR... I will make sure to capture it in an issue (so it is not lost).   also...  as discussed in a side conversation... we decided to create the supplied ns if it doesn't exist, on the premise that if the ns doesn't exist, the test would fail.  The main reason to not create a ns, is the user running kuttl may not have the rights to create a ns.  If the tests were going to fail anyway... attempting to create the ns will fail for another reason but if they have rights has the chance of succeeding.   This would change the nature of these checks... at min everything is potentially "auto-created" reducing the value of the suggested struct.  However perhaps there is room for something similar...  I'm with you!  self-documenting code is better... also localizing decisions is better.

Reply:
@porridge turns out... this looks really good with the stucts.  thank you sir!